setup.py

```diff
@@ -121,6 +122,8 @@ def main():
             "-DPYTHON3_LIBRARY=%s" % python_lib_path,
             "-DBUILD_opencv_python3=ON",
             "-DBUILD_opencv_python2=OFF",
+            # Disable the Java build by default as it is not needed
+            "-DBUILD_opencv_java=%s" % build_java,

```

Comment:
``get_build_env_var_by_name`` returns ``True`` or ``False``, CMake flags should be ``ON`` or ``OFF``.

Reply:
They take any boolean value (`1`, `0`, `ON`, `OFF`, `true`, `false`).  However, convention is ON/OFF, so I added logic to handle that.  I tested manually and observe the following:

By default, the generated cmake command line is as follows:
```
  Configuring Project
    Working directory:
      /tmp/pip-req-build-x8captf1/_skbuild/linux-x86_64-3.6/cmake-build
    Command:
      cmake /tmp/pip-req-build-x8captf1/opencv -G 'Unix Makefiles' -DCMAKE_INSTALL_PREFIX:PATH=/tmp/pip-req-build-x8captf1/_skbuild/linux-x86_64-3.6/cmake-install -DPYTHON_EXECUTABLE:FILEPATH=/export/apps/python/3.6.1/bin/python3.6 -DPYTHON_VERSION_STRING:STRING=3.6.1 -DPYTHON_INCLUDE_DIR:PATH=/export/apps/python/3.6.1/include/python3.6m -DPYTHON_LIBRARY:FILEPATH=/export/apps/python/3.6.1/lib/libpython3.6m.so -DSKBUILD:BOOL=TRUE -DCMAKE_MODULE_PATH:PATH=/tmp/pip-build-env-sg1jqlwf/overlay/lib/python3.6/site-packages/skbuild/resources/cmake -DPYTHON3_EXECUTABLE=/export/apps/python/3.6.1/bin/python3.6 -DPYTHON3_INCLUDE_DIR=/export/apps/python/3.6.1/include/python3.6m -DPYTHON3_LIBRARY=/export/apps/python/3.6.1/lib/libpython3.6m.so -DBUILD_opencv_python3=ON -DBUILD_opencv_python2=OFF -DBUILD_opencv_java=OFF -DOPENCV_SKIP_PYTHON_LOADER=ON -DOPENCV_PYTHON3_INSTALL_PATH=python -DINSTALL_CREATE_DISTRIB=ON -DBUILD_opencv_apps=OFF -DBUILD_SHARED_LIBS=OFF -DBUILD_TESTS=OFF -DBUILD_PERF_TESTS=OFF -DBUILD_DOCS=OFF -DCMAKE_BUILD_TYPE:STRING=Release
```

With `ENABLE_JAVA=1`, the command line is as follows:
```
  Configuring Project
    Working directory:
      /tmp/pip-req-build-sifkn0pp/_skbuild/linux-x86_64-3.6/cmake-build
    Command:
      cmake /tmp/pip-req-build-sifkn0pp/opencv -G 'Unix Makefiles' -DCMAKE_INSTALL_PREFIX:PATH=/tmp/pip-req-build-sifkn0pp/_skbuild/linux-x86_64-3.6/cmake-install -DPYTHON_EXECUTABLE:FILEPATH=/export/apps/python/3.6.1/bin/python3.6 -DPYTHON_VERSION_STRING:STRING=3.6.1 -DPYTHON_INCLUDE_DIR:PATH=/export/apps/python/3.6.1/include/python3.6m -DPYTHON_LIBRARY:FILEPATH=/export/apps/python/3.6.1/lib/libpython3.6m.so -DSKBUILD:BOOL=TRUE -DCMAKE_MODULE_PATH:PATH=/tmp/pip-build-env-6_18h_6z/overlay/lib/python3.6/site-packages/skbuild/resources/cmake -DPYTHON3_EXECUTABLE=/export/apps/python/3.6.1/bin/python3.6 -DPYTHON3_INCLUDE_DIR=/export/apps/python/3.6.1/include/python3.6m -DPYTHON3_LIBRARY=/export/apps/python/3.6.1/lib/libpython3.6m.so -DBUILD_opencv_python3=ON -DBUILD_opencv_python2=OFF -DBUILD_opencv_java=ON -DOPENCV_SKIP_PYTHON_LOADER=ON -DOPENCV_PYTHON3_INSTALL_PATH=python -DINSTALL_CREATE_DISTRIB=ON -DBUILD_opencv_apps=OFF -DBUILD_SHARED_LIBS=OFF -DBUILD_TESTS=OFF -DBUILD_PERF_TESTS=OFF -DBUILD_DOCS=OFF -DCMAKE_BUILD_TYPE:STRING=Release
```